src/Microsoft.DotNet.Darc/src/Darc/Operations/VirtualMonoRepo/ScanOperation.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Options.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DotNet.Darc.Operations.VirtualMonoRepo;
+
+internal class ScanOperation : Operation
+{
+    public ScanOperation(ScanCommandLineOptions options) : base(options, options.RegisterServices())
+    {
+    }
+
+    public override async Task<int> ExecuteAsync()
+    {
+        var vmrScanner = Provider.GetRequiredService<IVmrScanner>();
```

Comment:
It would be great if you added cancellation. You can have a look at `VmrOperationBase` where we use `CancellationKeyListener.ListenForCancellation` and pass the token down into the scan method so that we can cancel the process in the right places (e.g. between each repository)

Reply:
added